packages/paste-core/components/datagrid/stories/ComposableCells.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import {Box} from '@twilio-paste/box';
+import {Input} from '@twilio-paste/input';
+import {Menu, useMenuState, MenuButton, MenuItem, MenuSeparator} from '@twilio-paste/menu';
+import {MoreIcon} from '@twilio-paste/icons/esm/MoreIcon';
+import {Select, Option} from '@twilio-paste/select';
+import {DataGrid, DataGridHead, DataGridRow, DataGridHeader, DataGridBody, DataGridCell} from '../src';
+import {TableHeaderData, TableBodyData} from './constants';
+
+const ActionMenu: React.FC = () => {
+  const menu = useMenuState();
+  return (
+    <Box display="flex" justifyContent="center">
+      <MenuButton {...menu} variant="reset" size="reset">
+        <MoreIcon decorative={false} title="More options" />
+      </MenuButton>
+      <Menu {...menu} aria-label="Preferences">
+        <MenuItem {...menu}>Settings</MenuItem>
+        <MenuItem {...menu} disabled>
+          Extensions
+        </MenuItem>
+        <MenuSeparator {...menu} />
+        <MenuItem {...menu}>Keyboard shortcuts</MenuItem>
+      </Menu>
+    </Box>
+  );
+};
+
+export const ComposableCells: React.FC = () => {
+  /* eslint-disable react/no-array-index-key */
+  return (
+    <DataGrid aria-label="Simple datagrid">
+      <DataGridHead>
+        <DataGridRow>
+          <DataGridHeader data-testid="header-1">{TableHeaderData[0]}</DataGridHeader>
+          <DataGridHeader>{TableHeaderData[1]}</DataGridHeader>
+          <DataGridHeader>{TableHeaderData[2]}</DataGridHeader>
+          <DataGridHeader>{TableHeaderData[3]}</DataGridHeader>
+          <DataGridHeader>{TableHeaderData[4]}</DataGridHeader>
+          <DataGridHeader>Actions</DataGridHeader>
+        </DataGridRow>
+      </DataGridHead>
+      <DataGridBody>
+        {TableBodyData.map((row, rowIndex) => (
+          <DataGridRow key={`row-${rowIndex}`}>
+            {row.map((col, colIndex) => {
+              if (colIndex === 0 || colIndex === 1) {
+                return (
+                  <DataGridCell key={`col-${colIndex}`}>
```

Comment:
Would prefer to use `useUIDSeed` to create a UID seed and use those for the keys instead. In my experience that works super well for things like this, especially when a set is reordered, etc.

Reply:
While you can use `uid-library` for react keys, I actually recommend not doing so. This is because React keys aren't similar to HTML `id`s; they don't need to be globally unique. They only need to be contextually unique to this list. Furthermore it is too expensive to couple and call for something that a React implementation detail imo.